src/components/Providers/useAccountForProvider.spec.jsx

```diff
@@ -0,0 +1,179 @@
+import React from 'react'
+import { renderHook } from '@testing-library/react-hooks'
+
+import AppLike from 'test/AppLike'
+
+import useAccountForProvider from './useAccountForProvider'
+import { saveAccountToSettings } from './helpers'
+
+jest.mock('./helpers', () => ({
+  saveAccountToSettings: jest.fn()
+}))
+
+const accountsData = [{ _id: 'accountId01' }, { _id: 'accountId02' }]
+const settingsData = [{ account: { _id: 'accountId03' } }]
+
+const setup = ({ settings, accounts, selectedAccount } = {}) => {
+  const wrapper = ({ children }) => <AppLike>{children}</AppLike>
+
+  return renderHook(
+    () => useAccountForProvider(settings, accounts, selectedAccount),
+    {
+      wrapper
+    }
+  )
+}
+
+describe('useAccountForProvider', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('should return null if settings and accounts are loading', async () => {
+    const {
+      result: { current: account }
+    } = setup({
+      settings: { data: null, isLoading: true },
+      accounts: { data: null, isLoading: true },
+      selectedAccount: null
+    })
+
+    expect(account).toBe(null)
+  })
+
+  it('should return null if only accounts is loading', async () => {
+    const {
+      result: { current: account }
+    } = setup({
+      settings: { data: settingsData, isLoading: false },
+      accounts: { data: null, isLoading: true },
+      selectedAccount: null
+    })
+
+    expect(account).toBe(null)
+  })
+
+  it('should return null if only settings is loading', async () => {
+    const {
+      result: { current: account }
+    } = setup({
+      settings: { data: null, isLoading: true },
+      accounts: { data: null, isLoading: false },
+      selectedAccount: null
+    })
+
+    expect(account).toBe(null)
+  })
+
+  describe('for the first render, aka selectedAccount is null', () => {
+    const selectedAccount = null
+
+    it('should return the account from settings', async () => {
+      const {
+        result: { current: account }
+      } = setup({
+        settings: { data: settingsData, isLoading: false },
+        accounts: { data: null, isLoading: false },
+        selectedAccount
+      })
+
+      expect(account).toStrictEqual(settingsData[0].account)
```

Comment:
could we name this "accountFromSettingData = settingsData[0].account" in line 12 🙏 
This helps to find, which parameter is expected by the test